File: abc297/abc297_a.js

function main(input) {
	const args = input.split('\n');

	const num = args[0].split(' ');
	const N = parseInt(num[0]), D = parseInt(num[1]);
	const T = args[1].split(' ');

	for (let i = 0; i < N - 1; i++) {
		if (parseInt(T[i + 1]) - parseInt(T[i]) <= D) {
			console.log(T[i + 1]);
			return;
		}
	}

	console.log(-1);

}

main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: abc297/abc297_a.py

def main():
    N, D = map(int, input().split(" "))
    T = input().split(" ")

    for i in range(N-1):
        if int(T[i+1])-int(T[i]) <= D:
            print(T[i+1])
            exit(0)

    print(-1)


main()
